Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,88 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        {
+            int i, s = -1, q = -1, e = -1;
+            // char c;
+            for (i = 0; i < equation.Length; i += 1)
+            {
+                if (equation[i] == '*')
+                    s = i;
+                else if (equation[i] == '=')
+                    e = i;
+                else if (equation[i] == '?')
+                    q = i;
+            }
+            //Console.WriteLine($"{s}{e}{q}");
+            if (q < e)
+            {
+                string value = equation.Substring(e + 1, (equation.Length) - e - 1);
+                // Console.WriteLine($"{value}");
+                int val = int.Parse(value);
+                string ope1, ope2;
+                if (s < q)
+                {
+                    ope1 = equation.Substring(0, s);
+                    ope2 = equation.Substring(s + 1, e - s - 1);
+                }
+                else
+                {
+                    ope1 = equation.Substring(s + 1, e - s - 1);
+                    ope2 = equation.Substring(0, s);
+                }
+                /*Console.WriteLine($"{ope1}");
+                Console.WriteLine($"{ope2}");*/
+                int operand1 = int.Parse(ope1);
+                int answer = val / operand1;
+                if (val % operand1 != 0)
+                    return -1;
+                string answer1 = Convert.ToString(answer);
+                if (answer1.Length == ope2.Length)
+                {
+                    for (i = 0; i < ope2.Length; i += 1)
+                    {
+                        if (ope2[i] == '?')

```

Comment:
Try to use Single responsibility principle or divide the functionality in multiple functions.